tests/mocha/serializer_test.js

```diff
@@ -0,0 +1,1758 @@
+    'Zalgo',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">z̴̪͈̲̜͕̽̈̀͒͂̓̋̉̍a̸̧̧̜̻̘̤̫̱̲͎̞̻͆̋ļ̸̛̖̜̳͚̖͔̟̈́͂̉̀͑̑͑̎ǵ̸̫̳̽̐̃̑̚̕o̶͇̫͔̮̼̭͕̹̘̬͋̀͆̂̇̋͊̒̽</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.ControlChars = new SerializerTestCase(
+    'ControlChars',
+    '<xml xmlns="https://developers.google.com/blockly/xml">' +
+    '<block type="procedures_defreturn" id="id******************" x="42" y="42">' +
+    '<field name="NAME">&#x01;&#xa1;</field>' +
+    '</block>' +
+    '</xml>');
+Serializer.Mutations.Procedure.Names.testCases = [
+  Serializer.Mutations.Procedure.Names.Symbols,
+  Serializer.Mutations.Procedure.Names.EscapedSymbols,
+  Serializer.Mutations.Procedure.Names.SingleQuotes,
+  Serializer.Mutations.Procedure.Names.DoubleQuotes,
+  Serializer.Mutations.Procedure.Names.Numbers,
+  Serializer.Mutations.Procedure.Names.Emoji,
+  Serializer.Mutations.Procedure.Names.Russian,
+  Serializer.Mutations.Procedure.Names.Japanese,
+  Serializer.Mutations.Procedure.Names.Zalgo,
+  // TODO: Uncoment once #4945 is merged.
+  // Serializer.Mutations.Procedure.Names.ControlChars,
+];
+
+Serializer.Mutations.Procedure.testSuites = [
+  Serializer.Mutations.Procedure.Names,
+];
+
+Serializer.Mutations.testSuites = [
+  Serializer.Mutations.ControlsIf,
+  Serializer.Mutations.ListCreate,
+  Serializer.Mutations.Procedure,
+];
+
+Serializer.testSuites = [
+  Serializer.Attributes,
+  Serializer.Fields,
+  Serializer.Icons,
+  Serializer.Connections,
+  Serializer.Mutations,
+];
+
+var runSerializerTestSuite = (serializer, deserializer, testSuite) => {
+
+  const createTestFunction = function(test) {
+    return function() {
+      Blockly.Xml.domToWorkspace(
+          Blockly.Xml.textToDom(test.xml), this.workspace);
+      if (serializer && deserializer) {
+        const save = serializer(Blockly.serialization.save(this.workspace));
+        this.workspace.clear();
+        Blockly.serialization.load(deserializer(save), this.workspace);
+      }
+      var newXml = Blockly.Xml.workspaceToDom(this.workspace);
+      chai.assert.equal(Blockly.Xml.domToText(newXml), test.xml);
+    };
+  };
+
+  // This takes in a suite, but we don't care.
+  const createTestCaseFunction = function(_) {
+    return createTestFunction;
+  };
+
+  let suiteCall = (testSuite.skip ? suite.skip : suite);
+  suiteCall = (testSuite.only ? suite.only : suiteCall);
+
+  suiteCall(testSuite.title, function() {
+    setup(function() {
+      this.workspace = new Blockly.Workspace();
+    });
+
+    teardown(function() {
+      this.workspace.dispose();
+    });
+
+    testHelpers.runTestSuites(
+        testSuite.testSuites, createTestCaseFunction);
+    testHelpers.runTestCases(testSuite.testCases, createTestFunction);
+  });
+};
+
+runSerializerTestSuite(null, null, Serializer);
+Serializer.skip = true;
```

Comment:
Just curious why we are skipping here?

Reply:
Skipping because this work was done before deserialization of blocks was added. That happens in the next PR & this gets unskipped :D